icu4c/source/test/intltest/numbertest_range.cpp

```diff
@@ -910,6 +911,14 @@ void NumberRangeFormatterTest::testGetDecimalNumbers() {
     }
 }
 
+void NumberRangeFormatterTest::test21684_Performance() {
+    IcuTestErrorCode status(*this, "test21684_Performance");
+    LocalizedNumberRangeFormatter lnf = NumberRangeFormatter::withLocale("en");
+    // The following two lines of code should finish quickly.
+    lnf.formatFormattableRange({"-1e99999", status}, {"0", status}, status);
+    lnf.formatFormattableRange({"0", status}, {"1e99999", status}, status);
+}
```

Comment:
This does not actually generate an error if the execution time is long. Is the idea for anyone running the tests to just look at the logged execution time and do something if it seems too long? Maybe we should at least have a infoln stating that; but that would not affect CI tests. Maybe better to call uprv_getRawUTCtime() at the beginning and end of the test and generate an error if the difference is significantly greater than expected (i.e. 20 seconds?)

Reply:
The problem with checking clock time is that the clock time really differs a lot depending on the environment. Not only CPU speed, but also whether the library was built with ASAN, run with Valgrind, etc. So as of now I basically use this as a manual qualitative runtime test.

I can either:

1. Remove the test, since the bug is fixed
2. Keep the test and add a flaky clock time checker
3. Keep the test as proposed

Reply:
OK, I guess #3 seems best.